limesurvey/limesurvey.py

```diff
@@ -302,6 +305,10 @@ def _invoke(self, method: str, *params) -> dict | None:
         result = json_response.get("result")
 
         if isinstance(result, dict) and result.get("status") not in ("OK", None):
+            if result.get("status") == "Invalid session key":
+                self.set_session_key()
+                params.pop(0) # avoid infinite recursion
+                return self._invoke(method, *params)
             self.error_message = json_response.get("result").get("status")
         else:
             self.error_message = None
```

Comment:
What do you think about:

```
        json_response = response.json()
        result = json_response.get("result")
        self.error_message = None

        if not isinstance(result, dict):
            return json_response

        if result.get("status") == API_SESSION_KEY_EXPIRED_MESSAGE:
            self.set_session_key()
            # Pop session key to avoid infinite recursion
            params.pop(0)
            return self._invoke(method, *params)

        if result.get("status") not in ("OK", None):
            self.error_message = json_response.get("result").get("status")

```

Reply:
I haven't tested it. But it reads better than the nested if

Reply:
I tested it, and works. I will make the change.